lttoolbox/state.cc

```diff
@@ -532,56 +531,58 @@ State::filterFinals(map<Node *, double> const &finals,
 
   response = NFinals(response, max_analyses, max_weight_classes);
 
-  result = L"";
-  for(vector<pair<wstring, double>>::iterator it = response.begin(); it != response.end(); it++)
+  result.clear();
+  for(vector<pair<UString, double>>::iterator it = response.begin(); it != response.end(); it++)
   {
-    result += L'/';
+    result += '/';
     result += it->first;
     if(display_weights)
     {
-      result += L"<W:";
-      result += to_wstring(it->second);
-      result += L">";
+      UChar* temp = new UChar[16];
```

Comment:
That should just be `UChar temp[16]{};` - no need for an allocation. It could even write directly to the end of `result`, though that's rather uglier code so wouldn't recommend here.